cyipopt/tests/unit/test_scipy_optional.py

```diff
@@ -70,3 +70,102 @@ def test_minimize_ipopt_nojac_constraints_if_scipy():
     expected_res = np.array([1.001867, 0.99434067, 1.05070075, 1.17906312,
                              1.38103001])
     np.testing.assert_allclose(res.get("x"), expected_res)
+
+
+@pytest.mark.skipif("scipy" not in sys.modules,
+                    reason="Test only valid if Scipy available.")
+def test_minimize_ipopt_jac_and_hessians_constraints_if_scipy(
+):
+    """`minimize_ipopt` works with objective gradient and Hessian 
+       and constraint jacobians and Hessians."""
+    from scipy.optimize import rosen, rosen_der, rosen_hess
+    x0 = [0.0, 0.0]
+    constr = {
+        "type": "ineq",
+        "fun": lambda x: -x[0]**2 - x[1]**2 + 2,
+        "jac": lambda x: np.array([-2 * x[0], -2 * x[1]]),
+        "hess": lambda x, v: -2 * np.eye(2) * v[0]
+    }
+    bounds = [(-1.5, 1.5), (-1.5, 1.5)]
+    res = cyipopt.minimize_ipopt(rosen, x0, jac=rosen_der, hess=rosen_hess,
+                                 constraints=constr)
+    assert isinstance(res, dict)
+    assert np.isclose(res.get("fun"), 0.0)
+    assert res.get("status") == 0
+    assert res.get("success") is True
+    expected_res = np.array([1.0, 1.0])
+    np.testing.assert_allclose(res.get("x"), expected_res, rtol=1e-5)
+
+
+@pytest.mark.skipif("scipy" not in sys.modules,
+                    reason="Test only valid if Scipy available.")
+def test_minimize_ipopt_hs071():
+    """ `minimize_ipopt` works with objective gradient and Hessian
+        and constraint jacobians and Hessians. The objective and
+        the constraints functions return a tuple containing the
+        function value and the evaluated gradient or jacobian.
+    """
+    # Hock & Schittkowski test problem 71
+    #
+    # min x0*x3*(x0+x1+x2)+x2
+    #
+    # s.t. x0**2 + x1**2 + x2**2 + x3**2 - 40 = 0
+    #      x0 * x1 * x2 * x3 - 25 >= 0
+    #      1 <= x0,x1,x2,x3 <= 5
```

Comment:
Can you please refactor this docstring so that it's in standard Numpy format? Single line initial description, followed by anything else as the main body etc.

Reply:
Done.